patient-profile/app.R

```diff
@@ -1,42 +1,59 @@
 library(teal.modules.clinical)
 library(teal.modules.general)
-library(scda)
-library(scda.2022)
-library(nestcolor)
-
 options(shiny.useragg = FALSE)
 
 nest_logo <- "https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/PNG/nest.png"
 
-ADSL <- synthetic_cdisc_data("latest")$adsl
-ADMH <- synthetic_cdisc_data("latest")$admh
-ADAE <- synthetic_cdisc_data("latest")$adae
-ADCM <- synthetic_cdisc_data("latest")$adcm
-ADVS <- synthetic_cdisc_data("latest")$advs
-ADLB <- synthetic_cdisc_data("latest")$adlb
+## Data reproducible code ----
+data <- teal_data()
+data <- within(data, {
+  library(scda)
+  library(scda.2022)
+  library(nestcolor)
 
-## Modify ADCM
-ADCM$CMINDC <- paste0("Indication_", as.numeric(ADCM$CMDECOD))
-ADCM$CMDOSE <- 1
-ADCM$CMTRT <- ADCM$CMCAT
-ADCM$CMDOSU <- "U"
-ADCM$CMROUTE <- "CMROUTE"
-ADCM$CMDOSFRQ <- "CMDOSFRQ"
-ADCM$CMASTDTM <- ADCM$ASTDTM
-ADCM$CMAENDTM <- ADCM$AENDTM
+  ADSL <- synthetic_cdisc_data("latest")$adsl
+  ADMH <- synthetic_cdisc_data("latest")$admh
+  ADAE <- synthetic_cdisc_data("latest")$adae
+  ADCM <- synthetic_cdisc_data("latest")$adcm
+  ADVS <- synthetic_cdisc_data("latest")$advs
+  ADLB <- synthetic_cdisc_data("latest")$adlb
 
-teal.data::col_labels(
-  ADCM[c("CMINDC", "CMTRT", "ASTDY", "AENDY")]
-) <- c(
-  "Indication",
-  "Reported Name of Drug, Med, or Therapy",
-  "Study Day of Start of Medication",
-  "Study Day of End of Medication"
-)
+  ## Modify ADCM
+  ADCM$CMINDC <- paste0("Indication_", as.numeric(ADCM$CMDECOD))
+  ADCM$CMDOSE <- 1
+  ADCM$CMTRT <- ADCM$CMCAT
+  ADCM$CMDOSU <- "U"
+  ADCM$CMROUTE <- "CMROUTE"
+  ADCM$CMDOSFRQ <- "CMDOSFRQ"
+  ADCM$CMASTDTM <- ADCM$ASTDTM
+  ADCM$CMAENDTM <- ADCM$AENDTM
+
+  teal.data::col_labels(
+    ADCM[c("CMINDC", "CMTRT", "ASTDY", "AENDY")]
+  ) <- c(
+    "Indication",
+    "Reported Name of Drug, Med, or Therapy",
+    "Study Day of Start of Medication",
+    "Study Day of End of Medication"
+  )
+
+  ## Modify ADHM
+  ADMH[["MHDISTAT"]] <- "ONGOING"
+  teal.data::col_labels(ADMH[c("MHDISTAT")]) <- c("Status of Disease")
+})
+
+datanames <- c("ADSL", "ADMH", "ADAE", "ADCM", "ADVS", "ADLB")
+datanames(data) <- datanames
+join_keys(data) <- default_cdisc_join_keys[datanames]
```

Comment:
```suggestion
join_keys(data) <- default_cdisc_join_keys[datanames]

join_keys(data)[["ADAE"]][["ADMH"]] <- c("STUDYID", "USUBJID")
join_keys(data)[["ADAE"]][["ADCM"]] <- c("STUDYID", "USUBJID")
join_keys(data)[["ADAE"]][["ADVS"]] <- c("STUDYID", "USUBJID")
join_keys(data)[["ADAE"]][["ADLB"]] <- c("STUDYID", "USUBJID")
join_keys(data)[["ADCM"]][["ADMH"]] <- c("STUDYID", "USUBJID")
join_keys(data)[["ADCM"]][["ADVS"]] <- c("STUDYID", "USUBJID")
join_keys(data)[["ADCM"]][["ADLB"]] <- c("STUDYID", "USUBJID")
join_keys(data)[["ADVS"]][["ADMH"]] <- c("STUDYID", "USUBJID")
join_keys(data)[["ADVS"]][["ADLB"]] <- c("STUDYID", "USUBJID")
join_keys(data)[["ADLB"]][["ADMH"]] <- c("STUDYID", "USUBJID")
```
tm_g_pp_patient_timeline module was failing.

Reply:
Hmmm, I need to check it how it was before. If two datasets can be merged together by keys to their shared parent. Lemme think

Reply:
Old join key
```
> old_jk$get_join_keys()
A join_keys object containing foreign keys between 6 datasets:
ADSL: [STUDYID, USUBJID]
  <-- ADAE: [STUDYID, USUBJID]
  <-- ADMH: [STUDYID, USUBJID]
  <-- ADCM: [STUDYID, USUBJID]
  <-- ADVS: [STUDYID, USUBJID]
  <-- ADLB: [STUDYID, USUBJID]
ADAE: [STUDYID, USUBJID, ASTDTM, AETERM, AESEQ]
  --> ADSL: [STUDYID, USUBJID]
  <-> ADMH: [STUDYID, USUBJID]
  <-> ADCM: [STUDYID, USUBJID]
  <-> ADVS: [STUDYID, USUBJID]
  <-> ADLB: [STUDYID, USUBJID]
ADMH: [STUDYID, USUBJID, ASTDTM, MHSEQ]
  --> ADSL: [STUDYID, USUBJID]
  <-> ADAE: [STUDYID, USUBJID]
  <-> ADCM: [STUDYID, USUBJID]
  <-> ADVS: [STUDYID, USUBJID]
  <-> ADLB: [STUDYID, USUBJID]
ADCM: [STUDYID, USUBJID, ASTDTM, CMSEQ, ATC1CD, ATC2CD, ATC3CD, ATC4CD]
  --> ADSL: [STUDYID, USUBJID]
  <-> ADAE: [STUDYID, USUBJID]
  <-> ADMH: [STUDYID, USUBJID]
  <-> ADVS: [STUDYID, USUBJID]
  <-> ADLB: [STUDYID, USUBJID]
ADVS: [STUDYID, USUBJID, PARAMCD, AVISIT]
  --> ADSL: [STUDYID, USUBJID]
  <-> ADAE: [STUDYID, USUBJID]
  <-> ADMH: [STUDYID, USUBJID]
  <-> ADCM: [STUDYID, USUBJID]
  <-> ADLB: [STUDYID, USUBJID]
ADLB: [STUDYID, USUBJID, PARAMCD, AVISIT]
  --> ADSL: [STUDYID, USUBJID]
  <-> ADAE: [STUDYID, USUBJID]
  <-> ADMH: [STUDYID, USUBJID]
  <-> ADCM: [STUDYID, USUBJID]
  <-> ADVS: [STUDYID, USUBJID] 
```
New Join keys 
```
> join_keys(data) <- default_cdisc_join_keys[datanames]
> join_keys(data)
A join_keys object containing foreign keys between 6 datasets:
ADSL: [STUDYID, USUBJID]
  <-- ADAE: [STUDYID, USUBJID]
  <-- ADCM: [STUDYID, USUBJID]
  <-- ADLB: [STUDYID, USUBJID]
  <-- ADMH: [STUDYID, USUBJID]
  <-- ADVS: [STUDYID, USUBJID]
ADAE: [STUDYID, USUBJID, ASTDTM, AETERM, AESEQ]
  --> ADSL: [STUDYID, USUBJID]
ADCM: [STUDYID, USUBJID, ASTDTM, CMSEQ, ATC1CD, ATC2CD, ATC3CD, ATC4CD]
  --> ADSL: [STUDYID, USUBJID]
ADLB: [STUDYID, USUBJID, PARAMCD, AVISIT]
  --> ADSL: [STUDYID, USUBJID]
ADMH: [STUDYID, USUBJID, ASTDTM, MHSEQ]
  --> ADSL: [STUDYID, USUBJID]
ADVS: [STUDYID, USUBJID, PARAMCD, AVISIT]
  --> ADSL: [STUDYID, USUBJID] 
```

Reply:
Ok, we need to resolve it in teal.data then

Reply:
@kartikeyakirar Seems to be fixed. Thanks for catching this up early